### ISSUE TITLE
Colorize "expected" and "actual"

### ISSUE THREAD

**Issue by golopot** [stretchr/testify#946]
Can we have an option to colorize "expected" and "actual" in test output? Like this:
![Screenshot from 2020-05-13 18-27-59](https://user-images.githubusercontent.com/16285118/81802475-ba0fc100-9548-11ea-8d31-5a335b23981b.png)

![0_ZTerrO_akuDWSV8h](https://user-images.githubusercontent.com/16285118/81802442-a9f7e180-9548-11ea-9a74-2935bf245fb4.png)

For me the colors helps because it allows me to locate the diff faster. This kind of coloring is very common in Node.js test frameworks.

**Comment by boyan-soubachov** [comment]
It's an interesting idea. I'm slightly leaning towards keeping testify as simple as possible but ultimately we do it for the community. If there is strong support for this, I'd be happy to look into it :)

**Comment by mvdkleijn** [comment]
I like this idea. Perhaps the default should be set to the current behaviour and starting 2.0 we could enable it by default?

@boyan-soubachov I'd say let's do this assuming we can keep it a lightweight and KISSy solution.

Anyone any concrete ideas for a PR?

**Comment by arsham** [comment]
I recommend an interface for decorating the output. The default implementation behaves as is now without decorating the output. Then anyone could implement this interface for colourful output.

## Example

[Spec](https://github.com/sclevine/spec) library has [this](https://github.com/sclevine/spec/blob/1a3336281f177d8495942048c06fc46af8ab4d68/spec.go#L347..L357) interface for generating reports. [Here](https://github.com/arsham/dbtools/blob/master/dbtesting/mocha.go) and example implementation.

**Comment by boyan-soubachov** [comment]
Agreed with @arsham. I think it makes a lot more sense to provide the interface which users can customize. Not all users can see specific colours the same or well, not all colours have the same meanings to all people, etc.

**Comment by mvdkleijn** [comment]
So, since we're agreed that this is a nice idea... Here's an initial spec sheet for this new feature:

- [ ] Use an interface for decoration
- [ ] Default interface implementation should leave the output undecorated (pre 2.0 at least)
- [ ] An optional implementation should be provided for people who just want it to work
- [ ] Initial optional implementation should colourize actual vs expected at the very least

For now I don't think it should be more than this.

**Comment by alxn** [comment]
This will detect when running in interactive vs non-iterative mode (i.e. I don't want to see control characters in logged output).

**Comment by juanzolotoochin** [comment]
Are there plans to merge https://github.com/stretchr/testify/pull/1232?

**Comment by DennisTheMenace780** [comment]
Just here to say that I would also love a little bit of colour! It makes it a little bit easier for people's eyes to parse the information dump that they see on the screen. Seems like a good addition to the suite :+1:
